background.js

```diff
@@ -242,9 +243,11 @@ async function load_accounts() {
         return false;
     });
     if (!success) {
+        accounts.queried = false;
         return;
     } else if (accounts.registered.length == 0) {
         ssoLog("no accounts registered");
+        accounts.queried = false;
```

Comment:
In this case the broker responded correctly, but no accounts are registered. This VERY likely won't change over time as the current intune implementation requires a restart of the system.

If we set ` accounts.queried = false;` here, we will reach out to the broker on certain events which is not needed.

Reply:
I'm not worried about useless contacting. Fact is that even this change does not allow to recover from an initial glitch while obtaining the accounts. I just had to restart the addon to get out of that stuck state while the host was not restarted. More work need to fix this issue of the addon.

Reply:
But having no accounts is different from getting a "request failed". If we get a valid response without any accounts, this is still valid and we should accept it.

Reply:
There is really no point in forcing an addon restart just to update this state. That's just weird. The user may log in to intune later on if this had not happen before.

Reply:
At least with the official MS implementation or himmelblau a system restart is required after the (first) login. But if you are convinced that there are hypothetical providers that do not require this, we can keep it.

Reply:
How broken is that? Even for the sane implementation? Why that?

Reply:
In case of MS, the (first) Intune login is performed by the user after the user session start (and by that also after the start of the identity brokers). I don't know about the exact implementation, but according to our experience a system restart is needed to properly sync the state. It might be possible to also just restart a couple of systemd services.

In case of himmelblau, the intune login (Entra ID login) happens via a PAM module. By that, the user is already registered when entering the user session - and by that the user is always known when starting the browser.

Reply:
Let me drop that line then, just keeping the other after request failures. I still need to check if I can reproduce the issue I was seeing earlier today.